### PR TITLE
Add sample versions of standard deviation and variance functions

### LINKED ISSUE #57950: SQL: STDEV function support
Since https://github.com/elastic/elasticsearch/issues/49554 has been implemented, we should consider adding STDEV function support to SQL as well.

### PR DESCRIPTION
**Pull request by bpintea**
This PR adds the sampling variations of the standard deviation and variance agg functions, `STDDEV_SAMP` and `VAR_SAMP` respectively.

Closes #57950

### CODE REVIEW

**Comment by bpintea** [comment]
@elasticmachine update branch

File: x-pack/plugin/sql/qa/server/src/main/resources/docs/docs.csv-spec [hunk 1]
```diff
@@ -1615,6 +1637,28 @@ SELECT MIN(salary / 24.0) AS min, MAX(salary / 24.0) AS max, VAR_POP(salary / 24
...
+
+aggVarSampScalars
+schema::min:d|max:d|varsamp:d
+// tag::aggVarSampScalars
+SELECT MIN(salary / 24.0) AS min, MAX(salary / 24.0) AS max, VAR_SAMP(salary / 24.0) AS varsamp FROM emp;
+
+       min        |       max        |     varsamp
+------------------+------------------+----------------
+1055.1666666666667|3124.9583333333335|332278.830154847
+
```
Comment (matriv) [inline x-pack/plugin/sql/qa/server/src/main/resources/docs/docs.csv-spec:1658]: extra empty line

**Review by matriv — COMMENTED** [review]
Looks good, but can we have some unit tests in QueryTranslatorTests?
It would be great to show that when having those aggs on the same field we use ExtendedStats agg)

**Review by costin — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM
Agree with @matriv: a query translator test would make this feature complete.

**Comment by bpintea** [comment]
@elasticmachine update branch

**Review by matriv — APPROVED** [review]
LGTM, thx!

**Pull request merged**